backend/shared/shared/oauth2/views.py

```diff
@@ -0,0 +1,13 @@
+from django.conf import settings
+from django.shortcuts import redirect
+from django_auth_adfs.views import OAuth2CallbackView
+
+
+class HelsinkiOAuth2CallbackView(OAuth2CallbackView):
+    def get(self, request):
+        response = super().get(request)
+
+        if response.status_code == 302:
+            return redirect(settings.ADFS_LOGIN_REDIRECT_URL)
+        else:
+            return redirect(settings.ADFS_LOGIN_REDIRECT_URL_FAILURE)
```

Comment:
Should we log some info about the response if it's a failure, like response code? OAuth2CallbackView in django-auth-adfs uses 400, 401, 403 for separate error cases

Reply:
Yep, I'll add logging.

Reply:
Fixed